simulator/src/main/java/com/hedera/block/simulator/grpc/PublishStreamObserver.java

```diff
@@ -27,21 +30,31 @@
 public class PublishStreamObserver implements StreamObserver<PublishStreamResponse> {
 
     private final Logger logger = System.getLogger(getClass().getName());
+    private final AtomicBoolean streamEnabled;
 
-    /** Creates a new PublishStreamObserver instance. */
-    public PublishStreamObserver() {}
+    /** Creates a new PublishStreamObserver instance.
+     *
+     * @param streamEnabled is responsible for signaling, whether streaming should continue
+     */
+    public PublishStreamObserver(final AtomicBoolean streamEnabled) {
+        this.streamEnabled = streamEnabled;
+    }
 
     /** what will the stream observer do with the response from the server */
     @Override
     public void onNext(PublishStreamResponse publishStreamResponse) {
         logger.log(Logger.Level.INFO, "Received Response: " + publishStreamResponse.toString());
     }
 
-    /** what will the stream observer do when an error occurs */
+    /** Responsible for stream observer behaviour, in case of error. For now, we will stop the stream for every error. In the future we'd want to have a retry mechanism depending on the error. */
     @Override
-    public void onError(Throwable throwable) {
-        logger.log(Logger.Level.ERROR, "Error: " + throwable.toString());
-        // @todo(286) - handle the error
+    public void onError(@NonNull final Throwable streamError) {
+        streamEnabled.set(false);
+        Status status = Status.fromThrowable(streamError);
+        logger.log(
+                Logger.Level.ERROR,
+                "Error %s with status code %s.".formatted(status.getCode(), streamError),
```

Comment:
I think my suggestion had the values reversed, my apologies.
```suggestion
                "Error %s with status code %s.".formatted(streamError, status),
```

Reply:
You're right, thanks for noticing 👍